--- a/conf.d/000-general	1970-01-01 02:00:00.000000000 +0200
+++ b/conf.d/000-general	2016-05-19 02:30:01.012235028 +0300
@@ -0,0 +1,35 @@
+domain (ip ip6) table filter chain INPUT {
+    # port scans (module is in xtables-addons, requires dkms, compiling, etc in jessie...)
+    #mod psd psd-weight-threshold 21 psd-delay-threshold 300 psd-lo-ports-weight 3 psd-hi-ports-weight 1 DROP;
+}
+# unneeded icmp
+domain ip table filter chain INPUT proto icmp {
+    icmp-type address-mask-request DROP;
+    icmp-type timestamp-request DROP;
+}
+# ping flooding/smurf attack
+domain (ip ip6) table filter chain INPUT {
+    proto icmp mod limit limit 2/second ACCEPT;
+
+    proto tcp {
+        # extra bogus packets (beyond the already dropped INVALID ones)
+        tcp-flags ALL ALL DROP;
+        tcp-flags ALL NONE DROP;
+        tcp-flags (SYN FIN) (SYN FIN) DROP;
+        tcp-flags (SYN RST) (SYN RST) DROP;
+        tcp-flags (SYN URG) (SYN URG) DROP;
+
+        # syn check and flood-check
+        mod state state NEW {
+            ! syn DROP;
+            mod recent set name SYNFLOOD NOP;
+            mod recent update name SYNFLOOD seconds 1 hitcount 20 DROP;
+        }
+
+        # tcp reset attack
+        tcp-flags RST RST {
+            mod recent set name TCPRST NOP;
+            mod recent update name TCPRST seconds 1 hitcount 2 DROP;
+        }
+    }
+}
--- a/conf.d/001-lan	1970-01-01 02:00:00.000000000 +0200
+++ b/conf.d/001-lan	2016-05-19 03:12:55.624543345 +0300
@@ -0,0 +1,5 @@
+domain ip table filter chain INPUT {
+    # minimise chance of address spoofing impacting anything
+    saddr $RFC1918_SRC DROP;
+    daddr $RFC1918_DST DROP;
+}
--- a/conf.d/002-whitelist	1970-01-01 02:00:00.000000000 +0200
+++ b/conf.d/002-whitelist	2016-05-19 02:20:35.766577259 +0300
@@ -0,0 +1 @@
+domain (ip ip6) table filter chain INPUT saddr @ipfilter($WHITELISTED) ACCEPT;
--- a/conf.d/003-ssh	1970-01-01 02:00:00.000000000 +0200
+++ b/conf.d/003-ssh	2016-05-19 22:35:56.508678559 +0300
@@ -0,0 +1,4 @@
+domain (ip ip6) table filter chain INPUT proto tcp syn {
+    &RATELIMIT(ssh, (ssh), 300, 6);
+    mod multiport destination-ports (ssh) ACCEPT;
+}
--- a/ferm.conf	2014-10-26 00:52:35.000000000 +0300
+++ b/ferm.conf	2016-05-19 22:23:47.252691618 +0300
@@ -3,50 +3,51 @@
 #  Configuration file for ferm(1).
 #
 
-table filter {
-    chain INPUT {
-        policy DROP;
-
-        # connection tracking
-        mod state state INVALID DROP;
-        mod state state (ESTABLISHED RELATED) ACCEPT;
-
-        # allow local packet
-        interface lo ACCEPT;
-
-        # respond to ping
-        proto icmp ACCEPT; 
-
-        # allow IPsec
-        proto udp dport 500 ACCEPT;
-        proto (esp ah) ACCEPT;
+# NB:
+#   * When restarting ferm or fail2ban use the custom /usr/local/bin/fwadmin wrapper,
+#     because unless things are done in the right order, ferm flushes fail2ban's rules,
+#     then fail2ban logs "returned 100" errors about its missing rules, etc...
+
+@include 'vars.d/';
+@include 'funcs.d/';
+
+domain (ip ip6) {
+    table filter {
+
+        chain OUTPUT {
+            policy ACCEPT;
+
+            # connection tracking
+            #mod state state INVALID DROP;
+            mod state state (ESTABLISHED RELATED) ACCEPT;
+        }
+
+        chain FORWARD {
+            policy DROP;
+
+            # connection tracking
+            mod state state INVALID DROP;
+            mod state state (ESTABLISHED RELATED) ACCEPT;
+        }
+
+        chain INPUT {
+            policy DROP;
+
+            # connection tracking
+            mod state state INVALID DROP;
+            mod state state (ESTABLISHED RELATED) ACCEPT;
+
+            # allow local packet
+            interface lo ACCEPT;
+        }
 
-        # allow SSH connections
-        proto tcp dport ssh ACCEPT;
-    }
-    chain OUTPUT {
-        policy ACCEPT;
-
-        # connection tracking
-        #mod state state INVALID DROP;
-        mod state state (ESTABLISHED RELATED) ACCEPT;
-    }
-    chain FORWARD {
-        policy DROP;
-
-        # connection tracking
-        mod state state INVALID DROP;
-        mod state state (ESTABLISHED RELATED) ACCEPT;
     }
 }
 
-# IPv6:
-#domain ip6 {
-#    table filter {
-#        chain INPUT {
-#            policy ACCEPT;
-#            # ...
-#        }
-#        # ...
-#    }
+#domain arp {
 #}
+#
+#domain eb {
+#}
+
+@include 'conf.d/';
--- a/funcs.d/000-general	1970-01-01 02:00:00.000000000 +0200
+++ b/funcs.d/000-general	2016-05-19 02:21:09.902576648 +0300
@@ -0,0 +1,18 @@
+@def &TRIGGER($name, $trigger_ports, $trigger_protos, $target_ports, $target_protos, $duration) = @subchain @cat("TRIGGER_", $name) {
+    mod recent name @cat("TRIGGER_", $name) {
+        update seconds $duration proto $target_protos mod multiport destination-ports $target_ports ACCEPT;
+        set proto $trigger_protos mod multiport destination-ports $trigger_ports NOP;
+    }
+}
+
+@def &RATELIMIT($name, $ports, $duration, $count) = {
+    mod multiport destination-ports $ports @subchain @cat("RATELIM_", $name) {
+        mod recent name @cat("RATELIM_", $name) {
+            update seconds $duration hitcount $count @subchain @cat("RATELIM_", $name, "_DROP") {
+                LOG log-prefix "ferm-blocked: " log-level warning';
+                DROP;
+            }
+            set NOP;
+        }
+    }
+}
--- a/vars.d/000-general	1970-01-01 02:00:00.000000000 +0200
+++ b/vars.d/000-general	2016-05-19 02:20:39.134577199 +0300
@@ -0,0 +1,2 @@
+@def $RFC1918_SRC = (0.0.0.0/8 10.0.0.0/8 169.254.0.0/16 127.0.0.0/8 172.16.0.0/12 224.0.0.0/4 240.0.0.0/5);
+@def $RFC1918_DST = (0.0.0.0/8 224.0.0.0/4 239.255.255.0/24 240.0.0.0/5 255.255.255.255);
--- a/vars.d/001-lan	1970-01-01 02:00:00.000000000 +0200
+++ b/vars.d/001-lan	2016-05-19 03:10:12.440542721 +0300
@@ -0,0 +1 @@
+## nothing here yet
--- a/vars.d/002-whitelist	1970-01-01 02:00:00.000000000 +0200
+++ b/vars.d/002-whitelist	2016-05-20 01:01:39.756560821 +0300
@@ -0,0 +1 @@
+@def $WHITELISTED = (__WHITELISTADDRESSES__);
